packages/namers/default/src/DefaultNamer.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+import {Namer} from '@parcel/plugin';
+import crypto from 'crypto';
+import path from 'path';
+
+const COMMON_NAMES = new Set(['index', 'src', 'lib']);
+const DEFAULT_DIST_DIR = 'dist';
+
+export default new Namer({
+  name(bundle) {
+    // If the bundle has an explicit file path given (e.g. by a target), use that.
+    if (bundle.filePath) {
+      // TODO: what about multiple assets in the same dep?
+      // e.g. input is a Vue file, output is JS + CSS
+      // which is defined as a target in package.json?
+      return bundle.filePath;
+    }
+
+    // TODO: if split bundle, base name on original bundle names?
+    let entryAsset = bundle.assetGraph.getEntryAssets()[0];
+    let entryFilePath = entryAsset.filePath;
+    let name = path.basename(entryFilePath, path.extname(entryFilePath));
+
+    // If this is an entry bundle, exclude the hash and use the original relative path.
```

Comment:
What's the reasoning for this again?

Reply:
If you have `index.html` as an entry point, you typically want to keep that name rather than adding a content hash. Otherwise the link would change every time you deploy. Same thing for things like service workers, where the URL can never change.

Here is the original RFC: https://github.com/parcel-bundler/parcel/issues/872